File: csharp/DVSClient/DVSClient/address/format/enrichment/UKLocationComplete.cs
using DVSClient.Common;
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format.Enrichment
{
    public class UKLocationComplete
    {
        public double? Latitude { get; }
        public double? Longitude { get; }
        public GeocodeMatchLevel? MatchLevel { get; }
        public string Udprn { get; }
        public string Uprn { get; }
        public double? XCoordinate { get; }
        public double? YCoordinate { get; }

        public UKLocationComplete(RestApiEnrichmentUKLocationComplete api)
        {
            Latitude = api.Latitude;
            Longitude = api.Longitude;
            MatchLevel = api.MatchLevel?.GetEnumValueFromJsonName<GeocodeMatchLevel>();
            Udprn = api.Udprn ?? string.Empty;
            Uprn = api.Uprn ?? string.Empty;
            XCoordinate = api.XCoordinate;
            YCoordinate = api.YCoordinate;
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/address/format/enrichment/UKLocationComplete.java
package com.experian.dvs.client.address.format.enrichment;

import com.experian.dvs.client.address.format.GeocodeMatchLevel;
import com.experian.dvs.client.server.address.format.RestApiEnrichmentUKLocationComplete;

import java.util.Objects;

public class UKLocationComplete {

    private final Double latitude;
    private final Double longitude;
    private final GeocodeMatchLevel matchLevel;
    private final String udprn;
    private final String uprn;
    private final Double xCoordinate;
    private final Double yCoordinate;

    public UKLocationComplete(RestApiEnrichmentUKLocationComplete api) {
        this.latitude = api.getLatitude();
        this.longitude = api.getLongitude();
        this.matchLevel = api.getMatchLevel() != null ? GeocodeMatchLevel.fromValue(api.getMatchLevel()) : null;
        this.udprn = Objects.toString(api.getUdprn(), "");
        this.uprn = Objects.toString(api.getUprn(), "");
        this.xCoordinate = api.getXCoordinate();
        this.yCoordinate = api.getYCoordinate();
    }

    public Double getLatitude() {
        return latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public GeocodeMatchLevel getMatchLevel() {
        return matchLevel;
    }

    public String getUdprn() {
        return udprn;
    }

    public String getUprn() {
        return uprn;
    }

    public Double getXCoordinate() {
        return xCoordinate;
    }

    public Double getYCoordinate() {
        return yCoordinate;
    }
}
